### PR TITLE
Add ksql-tools

### PR DESCRIPTION
**Pull request by rodesai**
Add a module for ksql command line tools. Include a tool for
printing ksql metrics exposed as mbeans.

### CODE REVIEW

File: ksql-tools/src/main/java/io/confluent/ksql/tools/PrintMetrics.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      super(message);
+    }
+  }
+
+  private static void printMetrics(int pid) throws IOException {
+    // get all the jvms running locally and find the one we are interested in
+    Map<Integer, LocalVirtualMachine> jvms = LocalVirtualMachine.getAllVirtualMachines();
+    LocalVirtualMachine jvm = null;
+    if (pid == -1) {
+      for (LocalVirtualMachine j : jvms.values()) {
```
Comment (satybald) [inline ksql-tools/src/main/java/io/confluent/ksql/tools/PrintMetrics.java]: I'd propose to use the different variable name here. `i, j`s commonly used for index variables.

File: ksql-tools/src/main/java/io/confluent/ksql/tools/PrintMetrics.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+public class PrintMetrics {
+  public static void printHelp() {
+    System.err.println(
+        "usage: PrintMetrics "
+            + "[help] "
+            + "pid=<KSQL PID>");
+  }
+
+  public static class PrintMetricsException extends RuntimeException {
```
Comment (satybald) [inline ksql-tools/src/main/java/io/confluent/ksql/tools/PrintMetrics.java]: what do you think about having `PrintMetricsException` and `ArgumentMetricsException` in separate exception package?

**Comment by hjafarpour** [comment]
Would you like to add a shell script in `https://github.com/confluentinc/ksql/tree/master/bin` to run this?

File: ksql-tools/src/main/java/io/confluent/ksql/tools/printmetrics/PrintMetrics.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      }
+      MBeanAttributeInfo[] attributes = info.getAttributes();
+      for (MBeanAttributeInfo attributeInfo : attributes) {
+        Object attribute;
+        try {
+          attribute = connection.getAttribute(n, attributeInfo.getName());
+        } catch (Exception error) {
+          throw new PrintMetricsException("Unexpected error getting attribute " + error.getMessage());
+        }
+        System.out.println(attributeInfo.getName() + ": " + attribute);
```
Comment (bluemonk3y) [inline ksql-tools/src/main/java/io/confluent/ksql/tools/printmetrics/PrintMetrics.java]: at some point we might want to provide a formatter so data could displayed as CSV, JSON, statsd or delimited - the output could then be sent to a remote REST EP

Comment (rodesai) [reply ksql-tools/src/main/java/io/confluent/ksql/tools/printmetrics/PrintMetrics.java]: Missed this. I'll file an issue so we don't lose track of it.

**Pull request merged**